lib/spack/docs/developer_guide.rst

```diff
@@ -785,6 +785,32 @@ Unit tests
 Unit testing
 ------------
 
+---------------------
+Developer environment
```

Comment:
I'm not sure what the convention is for spack (@becker33, @tgamblin?) but I think we'll want to mark this very explicitly as unstable/experimental.  We're going to want to change it, almost certainly in breaking ways, and using it without knowing what you're doing could do anything from break your build to installing a very different version of a package than what shows up in the hash.

Maybe just put "Panic" on it in big, red, friendly letters?

Reply:
One big warning section coming up!